Mathlib/Order/Sublattice.lean

```diff
@@ -289,3 +289,90 @@ lemma map_top (f : LatticeHom α β) (h : Surjective f) : Sublattice.map f ⊤ =
   SetLike.coe_injective <| by simp [h.range_eq]
 
 end Sublattice
+
+namespace Sublattice
+variable {L M : Sublattice α} {f : LatticeHom α β} {s t : Set α} {a : α}
+
+/-- Binary product of sublattices as a sublattice. -/
+def prod (L : Sublattice α) (M : Sublattice β) : Sublattice (α × β) where
+  carrier := L ×ˢ M
+  supClosed' := L.supClosed.prod M.supClosed
+  infClosed' := L.infClosed.prod M.infClosed
+
+@[simp, norm_cast] lemma coe_prod (L : Sublattice α) (M : Sublattice β) :
+    L.prod M = (L ×ˢ M : Set (α × β)) := rfl
+
+@[simp] lemma mem_prod {M : Sublattice β} {p : α × β} : p ∈ L.prod M ↔ p.1 ∈ L ∧ p.2 ∈ M := Iff.rfl
+
+lemma prod_mono {L₁ L₂ : Sublattice α} {M₁ M₂ : Sublattice β} (hL : L₁ ≤ L₂) (hM : M₁ ≤ M₂) :
+    L₁.prod M₁ ≤ L₂.prod M₂ := Set.prod_mono hL hM
+
+lemma prod_mono_right {M₁ M₂ : Sublattice β} (hM : M₁ ≤ M₂) : L.prod M₁ ≤ L.prod M₂ :=
+  prod_mono le_rfl hM
+
+lemma prod_mono_left {L₁ L₂ : Sublattice α} {M : Sublattice β} (hL : L₁ ≤ L₂) :
+    L₁.prod M ≤ L₂.prod M := prod_mono hL le_rfl
+
+lemma prod_top (L : Sublattice α) : L.prod (⊤ : Sublattice β) = L.comap LatticeHom.fst :=
```

Comment:
Do you think this should be simp?

Reply:
Not so sure, which is why I didn't tag it

Reply:
Why not?